modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue

```diff
@@ -404,47 +462,101 @@ export default {
           requestMethod = apos.http.post;
 
           if (this.moduleName === '@apostrophecms/page') {
-            body._targetId = apos.page.page._id;
+            // New pages are always born as drafts
+            body._targetId = apos.page.page._id.replace(':published', ':draft');
             body._position = 'lastChild';
           }
         }
         let doc;
         try {
           doc = await requestMethod(route, {
             busy: true,
-            body
+            body,
+            draft: true
           });
           apos.bus.$emit('content-changed', doc);
         } catch (e) {
           if (e.body && (e.body.name === 'locked')) {
             await this.showLockedError(e);
             this.modal.showModal = false;
+            return;
           } else {
             await this.handleSaveError(e, {
               fallback: 'An error occurred saving the document.'
             });
             return;
           }
         }
+        if (andPublish) {
+          await this.publish(doc._id);
+        }
         this.$emit('modal-result', doc);
         this.modal.showModal = false;
         if ((this.moduleName === '@apostrophecms/page') && (!this.docId)) {
           window.location.href = doc._url;
         }
       });
     },
-    async getNewInstance () {
+    async publish(_id) {
+      try {
+        await apos.http.post(`${this.moduleAction}/${_id}/publish`, {
+          body: {},
+          busy: true
+        });
+        const event = {
+          name: 'revert-published-to-previous',
+          data: {
+            action: this.moduleAction,
+            _id
+          }
+        };
+        apos.notify(`Your changes have been published. <button data-apos-bus-event='${JSON.stringify(event)}'>Undo Publish</a>`, {
```

Comment:
Thanks for the closing tag catch. See above re: a followup ticket/design discussion/etc. if we want to change the approach in the beta timeframe (I'm not convinced it's wrong but it was just Stu and I and it could be a team discussion).